File: LevelOrderTraversal.py
class TreeNode:
    def __init__(self, value):
        self.val = value
        self.left = None
        self.right = None

def levelOrder(root):
    if not root:
        return []

    result = []
    queue = [root]

    while queue:
        level = []
        next_queue = []

        for node in queue:
            level.append(node.val)

            if node.left:
                next_queue.append(node.left)
            if node.right:
                next_queue.append(node.right)

        result.append(level)
        queue = next_queue

    return result

 
def printLevelOrder(result):
    for level in result:
        print(level)

 
root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)
root.left.left = TreeNode(4)
root.left.right = TreeNode(5)
root.right.left = TreeNode(6)
root.right.right = TreeNode(7)

 
result = levelOrder(root)

 
print("Level Order Traversal:")
printLevelOrder(result)

File: Program's_Contributed_By_Contributors/C++ Programs/Binary Tree/level-order-traversal.cpp
#include<iostream>
#include<queue>
#define endl "\n"
using namespace std;


// Basic Structure of a Tree
struct Node{
    int data;
    struct Node *left;
    struct Node *right;

    Node(int val){
        data = val;
        left = NULL;
        right = NULL;
    }
};

/* Level order Traversal of a Binary Tree using Queue */
/* Complexity : O(n) */

// Here NULLs are used to keep track of levels

void level_order(Node* root){
    if(root == NULL) 
     return;

    queue<Node*> q;
    q.push(root); q.push(NULL);

    while(!q.empty()){
        Node* node = q.front();
        q.pop();
        
        if(node != NULL){
            cout<<node->data<<" ";
            if(node->left)
                q.push(node->left);
            
            if(node->right)
                q.push(node->right);
            
        }else if(!q.empty()){
            q.push(NULL);
        }
    }
    cout<<endl;
}


int main(){
    Node* root = new Node(1);
    root->left = new Node(2);
    root->right = new Node(3);
    root->left->left = new Node(4);
    root->left->right = new Node(5);
    root->right->right = new Node(7);

    /*                       
                    1
                   / \
                  2   3
                 / \   \
                4   5   7    
        Expected-> level-order :  1,2,3,4,5,7
                   
    
    */
   cout<<"Level Order is : ";
   level_order(root);

    return 0;
}